src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,290 @@
+    }
+
+    std::string bank_shares = reinterpret_cast<char const *>(
+        sqlite3_column_text (b_select_shares_stmt, 0));
+
+    // initialize a weighted tree node
+    node = std::make_shared<weighted_tree_node_t> (parent_bank,
+                                                   bank_name,
+                                                   false,
+                                                   std::stoll (bank_shares),
+                                                   0);
+
+    // if there is no parent bank, then the root pointer points to the root bank
+    if (!parent_bank) {
+        root_ptr = node;
+    } else {
+        parent_bank->add_child (node);
+    }
+
+    rc = sqlite3_bind_text (b_select_sub_banks_stmt,
+                            1,
+                            bank_name.c_str (),
+                            -1,
+                            NULL);
+    if (rc != SQLITE_OK) {
+        fprintf (stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+    rc = sqlite3_step (b_select_sub_banks_stmt);
+    if (rc == SQLITE_ERROR) {
+        fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+
+    // vector of strings to hold sub banks
+    std::vector<std::string> banks;
+
+    // we've reached a bank with no sub banks, so print
+    // out all associations under this sub bank
+    if (rc != SQLITE_ROW) {
+        rc = sqlite3_bind_text (b_select_associations_stmt,
+                                1,
+                                bank_name.c_str (),
+                                -1,
+                                NULL);
+        if (rc != SQLITE_OK) {
+            fprintf(stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+
+        // execute SQL statement
+        rc = sqlite3_step (b_select_associations_stmt);
+        while (rc == SQLITE_ROW) {
+            std::string username = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 0));
+            std::string user_shares = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 1));
+            std::string user_job_usage = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 3));
+
+            // add user as a child of the node
+            user_node = std::make_shared<weighted_tree_node_t> (
+                            parent_bank,
+                            username,
+                            true,
+                            std::stoll (user_shares),
+                            std::stoll (user_job_usage));
+            node->add_child (user_node);
+            leaf_bank_usage += std::stoi (user_job_usage);
+
+            rc = sqlite3_step (b_select_associations_stmt);
+        }
```

Comment:
After the `while` loop exists, it might be worth checking that `rc == SQLITE_DONE` (or similar).  A sqlite error would also break the while loop but would go uncaught and without an error message.